Feature/rdk 56050

varatharajan568 opened the pull request:
combined changes of Externals library, SecManager separation, MW Lib seperation & L1 MW Testcase seperations, gst-rdk plugin

Deepikasri1234 committed:
RDK-56050 Merging of OCDM, gst-plugins-rdk into Library - Design

Reason for change: Migration to dev_sprint_25_2 from 117308 (patchset 12)
Test Procedure: Refer Ticket

Risks: Low

Priority: P1

Change-Id: If0a31d066478806bae29f09ab829b1108d7856ba
Signed-off-by: Deepikasri N <Deepikasri_N@Comcast.Com

Deepikasri1234 committed:
RDK-56050 Merging of OCDM, gst-plugins-rdk into Library - Design

Reason for change: Migration to dev_sprint_25_2 from 117308 (patchset 12)
Test Procedure: Refer Ticket

Risks: Low

Priority: P1

Change-Id: If0a31d066478806bae29f09ab829b1108d7856ba
Signed-off-by: Deepikasri N <Deepikasri_N@Comcast.Com

dp0000 committed:
Delete AampConfig.cpp.rej

removed .rej

dp0000 committed:
Delete aampgstplayer.cpp.rej

dp0000 committed:
Delete CMakeLists.txt.rej

Deepikasri1234 committed:
Merge branch 'feature/RDK-56050' of https://github.com/rdkcentral/aamp into feature/RDK-56050

Deepikasri1234 committed:
RDK-56050 Merging of OCDM, gst-plugins-rdk into Library - Design

Reason for change: for dev_Sprint_25_2
Test Procedure: Refer Ticket

Risks: Low

Priority: P1

Signed-off-by: Deepikasri N <Deepikasri_N@Comcast.Com

Deepikasri1234 committed:
RDK-56050 Merging of OCDM, gst-plugins-rdk into Library - Design

Reason for change: for dev_Sprint_25_2
Test Procedure: Refer Ticket

Risks: Low

Priority: P1

Signed-off-by: Deepikasri N <Deepikasri_N@Comcast.Com

Deepikasri1234 committed:
RDK-56050 Merging of OCDM, gst-plugins-rdk into Library - Design

Reason for change: for dev_Sprint_25_2
Test Procedure: Refer Ticket

Risks: Low

Priority: P1

Change-Id: If0a31d066478806bae29f09ab829b1108d7856ba
Signed-off-by: Deepikasri N <Deepikasri_N@Comcast.Com

Deepikasri1234 committed:
RDK-56050 Merging of OCDM, gst-plugins-rdk into Library - Design

Reason for change: for dev_Sprint_25_2
Test Procedure: Refer Ticket

Risks: Low

Priority: P1

Change-Id: If0a31d066478806bae29f09ab829b1108d7856ba
Signed-off-by: Deepikasri N <Deepikasri_N@Comcast.Com

Deepikasri1234 committed:
RDK-56050 Merging of OCDM, gst-plugins-rdk into Library - Design

Reason for change: for dev_Sprint_25_2
Test Procedure: Refer Ticket

Risks: Low

Priority: P1

Change-Id: If0a31d066478806bae29f09ab829b1108d7856ba
Signed-off-by: Deepikasri N <Deepikasri_N@Comcast.Com

Deepikasri1234 committed:
Merge remote-tracking branch 'origin/dev_sprint_25_2' into feature/RDK-56050

Deepikasri1234 committed:
RDK-56050 Merging of OCDM, gst-plugins-rdk into Library - Design

Reason for change: for dev_Sprint_25_2
Test Procedure: Refer Ticket

Risks: Low

Priority: P1

Change-Id: If0a31d066478806bae29f09ab829b1108d7856ba
Signed-off-by: Deepikasri N <Deepikasri_N@Comcast.Com

Deepikasri1234 committed:
RDK-56050 Merging of OCDM, gst-plugins-rdk into Library - Design

Reason for change: for dev_Sprint_25_2
Test Procedure: Refer Ticket

Risks: Low

Priority: P1

Change-Id: If0a31d066478806bae29f09ab829b1108d7856ba
Signed-off-by: Deepikasri N <Deepikasri_N@Comcast.Com

Deepikasri1234 committed:
RDK-56050 Merging of OCDM, gst-plugins-rdk into Library - Design

Reason for change: for dev_Sprint_25_2
Test Procedure: Refer Ticket

Risks: Low

Priority: P1

Change-Id: If0a31d066478806bae29f09ab829b1108d7856ba
Signed-off-by: Deepikasri N <Deepikasri_N@Comcast.Com

Deepikasri1234 committed:
RDK-56050 Merging of OCDM, gst-plugins-rdk into Library - Design

Reason for change: for dev_Sprint_25_2
Test Procedure: Refer Ticket

Risks: Low

Priority: P1

Change-Id: If0a31d066478806bae29f09ab829b1108d7856ba
Signed-off-by: Deepikasri N <Deepikasri_N@Comcast.Com

Deepikasri1234 committed:
Merge branch 'develop' into feature/RDK-56050

Deepikasri1234 committed:
RDK-56050 Merging of OCDM, gst-plugins-rdk into Library - Design

Reason for change: for dev_Sprint_25_2
Test Procedure: Refer Ticket

Risks: Low

Priority: P1

Change-Id: If0a31d066478806bae29f09ab829b1108d7856ba
Signed-off-by: Deepikasri N <Deepikasri_N@Comcast.Com

Deepikasri1234 committed:
Merge branch 'dev_sprint_25_2' into feature/RDK-56050

Deepikasri1234 committed:
RDK-56050 Merging of OCDM, gst-plugins-rdk into Library - Design

Reason for change: for dev_Sprint_25_2
Test Procedure: Refer Ticket

Risks: Low

Priority: P1

Change-Id: If0a31d066478806bae29f09ab829b1108d7856ba
Signed-off-by: Deepikasri N <Deepikasri_N@Comcast.Com

Deepikasri1234 committed:
RDK-56050 Merging of OCDM, gst-plugins-rdk into Library - Design

Reason for change: for dev_Sprint_25_2
Test Procedure: Refer Ticket

Risks: Low

Priority: P1

Change-Id: If0a31d066478806bae29f09ab829b1108d7856ba
Signed-off-by: Deepikasri N <Deepikasri_N@Comcast.Com

Deepikasri1234 committed:
RDK-56050 Merging of OCDM, gst-plugins-rdk into Library - Design

Reason for change: for dev_Sprint_25_2
Test Procedure: Refer Ticket

Risks: Low

Priority: P1

Change-Id: If0a31d066478806bae29f09ab829b1108d7856ba
Signed-off-by: Deepikasri N <Deepikasri_N@Comcast.Com

Deepikasri1234 committed:
RDK-56050 Merging of OCDM, gst-plugins-rdk into Library - Design

Reason for change: for dev_Sprint_25_2
Test Procedure: Refer Ticket

Risks: Low

Priority: P1

Change-Id: If0a31d066478806bae29f09ab829b1108d7856ba
Signed-off-by: Deepikasri N <Deepikasri_N@Comcast.Com

Deepikasri1234 committed:
RDK-56050 Merging of OCDM, gst-plugins-rdk into Library - Design

Reason for change: for dev_Sprint_25_2
Test Procedure: Refer Ticket

Risks: Low

Priority: P1

Change-Id: If0a31d066478806bae29f09ab829b1108d7856ba
Signed-off-by: Deepikasri N <Deepikasri_N@Comcast.Com

Deepikasri1234 committed:
RDK-56050 Merging of OCDM, gst-plugins-rdk into Library - Design

Reason for change: for dev_Sprint_25_2
Test Procedure: Refer Ticket

Risks: Low

Priority: P1

Change-Id: If0a31d066478806bae29f09ab829b1108d7856ba
Signed-off-by: Deepikasri N <Deepikasri_N@Comcast.Com

Deepikasri1234 committed:
Merge branch 'dev_sprint_25_2' into feature/RDK-56050

Deepikasri1234 committed:
RDK-56050 Merging of OCDM, gst-plugins-rdk into Library - Design

Reason for change: for dev_Sprint_25_2
Test Procedure: Refer Ticket

Risks: Low

Priority: P1

Change-Id: If0a31d066478806bae29f09ab829b1108d7856ba
Signed-off-by: Deepikasri N <Deepikasri_N@Comcast.Com

Deepikasri1234 committed:
RDK-56050 Merging of OCDM, gst-plugins-rdk into Library - Design

Reason for change: for dev_Sprint_25_2
Test Procedure: Refer Ticket

Risks: Low

Priority: P1

Change-Id: If0a31d066478806bae29f09ab829b1108d7856ba
Signed-off-by: Deepikasri N <Deepikasri_N@Comcast.Com

Deepikasri1234 committed:
RDK-56050 Merging of OCDM, gst-plugins-rdk into Library - Design

Reason for change: for dev_Sprint_25_2
Test Procedure: Refer Ticket

Risks: Low

Priority: P1

Change-Id: If0a31d066478806bae29f09ab829b1108d7856ba
Signed-off-by: Deepikasri N <Deepikasri_N@Comcast.Com

Commit:
DISABLE_SECURITY_TOKEN secmanager support

rekhap2kandhavelan committed:
Update L1-tests.yml

rkandh015 committed:
Rebase with dev_sprint_25_2

rkandh015 committed:
rebase with dev_sprint

rekhap2kandhavelan committed:
Update L1-tests.yml

rekhap2kandhavelan committed:
Update L1-tests.yml

Commit:
MonitorAVState avposition data type change

rkandh015 committed:
rebase with dev_sprint

Change-Id: I77d0d29b44a01ae8d9fb3ddd455815700a3dbc76

narenr94 committed:
dev_sprint_25_2 sync

narenr94 committed:
Merge remote-tracking branch 'origin/dev_sprint_25_2' into feature/RDK-56050

narenr94 committed:
build error fix 03

rkandh015 committed:
Resolve L1 build failure for aamp and middleware

Deepikasri1234 committed:
Merge branch 'dev_sprint_25_2' into feature/RDK-56050

Deepikasri1234 committed:
resolved compilation issues

Deepikasri1234 committed:
RDK-56050 Merging of OCDM, gst-plugins-rdk into Library - Design
Reason for change: comp issues
Test Procedure: Refer Ticket

Risks: Low

Priority: P1

Change-Id: If0a31d066478806bae29f09ab829b1108d7856ba
Signed-off-by: Deepikasri N <Deepikasri_N@Comcast.Com

Deepikasri1234 committed:
Merge branch 'dev_sprint_25_2' into feature/RDK-56050

rkandh015 committed:
removed cjson installation explicitly hence resolved in VPLAY-9947

Deepikasri1234 committed:
RDK-56050 Merging of OCDM, gst-plugins-rdk into Library - Design
Reason for change: adressing review comments
Test Procedure: Refer Ticket

Risks: Low

Priority: P1

Change-Id: If0a31d066478806bae29f09ab829b1108d7856ba
Signed-off-by: Deepikasri N <Deepikasri_N@Comcast.Com

narenr94 committed:
move Plugin Tests

narenr94 committed:
impl CMAKE_SIM_DRM_SUPPORT

dp0000 committed:
Merge branch 'dev_sprint_25_2' into feature/RDK-56050

Deepikasri1234 committed:
RDK-56050 Merging of OCDM, gst-plugins-rdk into Library - Design
Reason for change: Addressing review comments given by vinish
Test Procedure: Refer Ticket

Risks: Low

Priority: P1

Change-Id: If0a31d066478806bae29f09ab829b1108d7856ba
Signed-off-by: Deepikasri N <Deepikasri_N@Comcast.Com

Deepikasri1234 committed:
resolved compiltion failures

Deepikasri1234 committed:
profiling update in plugin code

Deepikasri1234 committed:
debug crush in gst-plugins

Deepikasri1234 committed:
debugging crash issue in peacock

Deepikasri1234 committed:
addressed build issues

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/rdkcentral/aamp?pullRequest=369) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/rdkcentral/aamp?pullRequest=369) before we can accept your contribution.<br/>**1** out of **6** committers have signed the CLA.<br/><br/>:white_check_mark: narenr94<br/>:x: dp0000<br/>:x: Deepikasri1234<br/>:x: rekhap2kandhavelan<br/>:x: rkandh015<br/>:x: Naren Ramesh<br/><hr/>**Naren Ramesh** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/rdkcentral/aamp?pullRequest=369) it.</sub>

rdkcmf-jenkins commented:
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
f729b73ba6db645adda7d67798226805
PRESENT
-->

# Coverity Issue - Variable copied when it could be moved
"Identifier" is copied and then passed-by-reference as parameter to STL insertion function "std::vector<GstPlayerPriv::CallbackData, std::allocator<GstPlayerPriv::CallbackData> >::push_back(std::vector<GstPlayerPriv::CallbackData, std::allocator<GstPlayerPriv::CallbackData> >::value_type const &)", when it could be moved instead.

_Low Impact, CWE-none_
_COPY_INSTEAD_OF_MOVE_

## How to fix
 Use "std::move(""Identifier"")" instead of "Identifier".

## Issue location
This issue was discovered outside the diff for this Pull Request. You can find it at:
[middleware/InterfacePlayerRDK.cpp:1665](undefined//blob/undefined/middleware/InterfacePlayerRDK.cpp#L1665)

rdkcmf-jenkins commented:
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
bfeb9fbf76938ea50aad4e532d49c476
PRESENT
-->

# Coverity Issue - Variable copied when it could be moved
"eventHandle" is passed-by-value as parameter to "std::shared_ptr<DrmMetaDataEvent>::shared_ptr(std::shared_ptr<DrmMetaDataEvent> const &) /*explicit =default*/", when it could be moved instead.

_Low Impact, CWE-none_
_COPY_INSTEAD_OF_MOVE_

## How to fix
 Use "std::move(""eventHandle"")" instead of "eventHandle".

## Issue location
This issue was discovered outside the diff for this Pull Request. You can find it at:
[drm/AampDRMLicManager.cpp:500](undefined//blob/undefined/drm/AampDRMLicManager.cpp#L500)

rdkcmf-jenkins commented:
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
e1973772a1247808604996469350d8c9
PRESENT
-->

# Coverity Issue - Variable copied when it could be moved
"keyId" is passed-by-value as parameter to "std::vector<unsigned char, std::allocator<unsigned char> >::vector(std::vector<unsigned char, std::allocator<unsigned char> > const &)", when it could be moved instead.

_Low Impact, CWE-none_
_COPY_INSTEAD_OF_MOVE_

## How to fix
 Use "std::move(""keyId"")" instead of "keyId".

## Issue location
This issue was discovered outside the diff for this Pull Request. You can find it at:
[middleware/drm/DrmSessionManager.cpp:459](undefined//blob/undefined/middleware/drm/DrmSessionManager.cpp#L459)

rdkcmf-jenkins commented:
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
ae74cb3f691c4d929df09df822b38bb9
PRESENT
-->

# Coverity Issue - Variable copied when it could be moved
"drmHelper" is passed-by-value as parameter to "std::shared_ptr<DrmHelper>::shared_ptr(std::shared_ptr<DrmHelper> const &) /*explicit =default*/", when it could be moved instead.

_Low Impact, CWE-none_
_COPY_INSTEAD_OF_MOVE_

## How to fix
 Use "std::move(""drmHelper"")" instead of "drmHelper".

## Issue location
This issue was discovered outside the diff for this Pull Request. You can find it at:
[drm/AampDRMLicManager.cpp:500](undefined//blob/undefined/drm/AampDRMLicManager.cpp#L500)

rdkcmf-jenkins commented:
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
8cd2817a0a0836708f72a10559f32b40
PRESENT
-->

# Coverity Issue - Use of auto that causes a copy
Using the "auto" keyword without an "&" causes the copy of an object of type "ContentSecurityManagerSession".

_Low Impact, CWE-none_
_AUTO_CAUSES_COPY_

## How to fix
 Use "const auto &" instead of "auto".

## Issue location
This issue was discovered outside the diff for this Pull Request. You can find it at:
[middleware/drm/DrmSessionManager.cpp:505](undefined//blob/undefined/middleware/drm/DrmSessionManager.cpp#L505)

rdkcmf-jenkins commented:
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
27cef176fa117c014eda23352d83b928
PRESENT
-->

# Coverity Issue - Variable copied when it could be moved
"licenseServerProxy" is passed-by-value as parameter to "std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::basic_string(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const &)", when it could be moved instead.

_Low Impact, CWE-none_
_COPY_INSTEAD_OF_MOVE_

## How to fix
 Use "std::move(""licenseServerProxy"")" instead of "licenseServerProxy".

## Issue location
This issue was discovered outside the diff for this Pull Request. You can find it at:
[drm/AampDRMLicManager.cpp:385](undefined//blob/undefined/drm/AampDRMLicManager.cpp#L385)

rdkcmf-jenkins commented:
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
ab9b1652361544f965acdf8f3592e80f
PRESENT
-->

# Coverity Issue - Variable copied when it could be moved
"licenseResponse" is passed-by-value as parameter to "std::shared_ptr<DrmData>::shared_ptr(std::shared_ptr<DrmData> const &) /*explicit =default*/", when it could be moved instead.

_Low Impact, CWE-none_
_COPY_INSTEAD_OF_MOVE_

## How to fix
 Use "std::move(""licenseResponse"")" instead of "licenseResponse".

## Issue location
This issue was discovered outside the diff for this Pull Request. You can find it at:
[drm/AampDRMLicManager.cpp:500](undefined//blob/undefined/drm/AampDRMLicManager.cpp#L500)

rdkcmf-jenkins commented:
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
84f7f23d6cde983b99d3800e05c2776c
PRESENT
-->

# Coverity Issue - Unchecked return value
Calling "wait_for" without checking return value (as is done elsewhere 12 out of 13 times).

_Medium Impact, CWE-252_
_CHECKED_RETURN_



## Issue location
This issue was discovered outside the diff for this Pull Request. You can find it at:
[middleware/InterfacePlayerRDK.cpp:2795](undefined//blob/undefined/middleware/InterfacePlayerRDK.cpp#L2795)

rdkcmf-jenkins commented on drm/AampDRMLicManager.cpp:
```diff
@@ -355,11 +396,11 @@ KeyState AampDRMLicenseManager::acquireLicense(std::shared_ptr<DrmHelper> drmHel
 	if (code == KEY_PENDING)
 	{
-		code = handleLicenseResponse(drmHelper, sessionSlot, cdmError, httpResponseCode, httpExtendedStatusCode, licenseResponse, eventHandle,  isLicenseRenewal);
+		code = handleLicenseResponse(responseCode, drmHelper, sessionSlot, cdmError, httpResponseCode, httpExtendedStatusCode, licenseResponse, eventHandle,  isLicenseRenewal);
```
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
d30410cdefed6080410d1fec988335cc
NOT_PRESENT
-->

Coverity issue no longer present as of: undefined
<details>
<summary>Show issue</summary>


# Coverity Issue - Variable copied when it could be moved
"licenseResponse" is passed-by-value as parameter to "std::shared_ptr<DrmData>::shared_ptr(std::shared_ptr<DrmData> const &) /*explicit =default*/", when it could be moved instead.

_Low Impact, CWE-none_
_COPY_INSTEAD_OF_MOVE_

## How to fix
 Use "std::move(""licenseResponse"")" instead of "licenseResponse".

</details>

Vinish100 replied:
fix this

dp0000 replied:
fixed

rdkcmf-jenkins commented on drm/AampDRMLicManager.cpp:
```diff
@@ -50,31 +50,70 @@
-    instance->RegisterLicenseDataCb([_this](std::shared_ptr<DrmHelper> drmHelper, int sessionSlot, int &cdmError, int streamType,void *metaDataPtr, bool isLicenseRenewal = false ) -> KeyState {
-        return _this->acquireLicense(drmHelper, sessionSlot, cdmError,
+    instance->RegisterLicenseDataCb([_this](int &responseCode, std::shared_ptr<DrmHelper> drmHelper, int sessionSlot, int &cdmError, int streamType,void *metaDataPtr, bool isLicenseRenewal = false ) -> KeyState {
+        return _this->acquireLicense(responseCode, drmHelper, sessionSlot, cdmError,
```
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
398e7589e469827525758a9c4292cac3
NOT_PRESENT
-->

Coverity issue no longer present as of: undefined
<details>
<summary>Show issue</summary>


# Coverity Issue - Variable copied when it could be moved
"drmHelper" is passed-by-value as parameter to "std::shared_ptr<DrmHelper>::shared_ptr(std::shared_ptr<DrmHelper> const &) /*explicit =default*/", when it could be moved instead.

_Low Impact, CWE-none_
_COPY_INSTEAD_OF_MOVE_

## How to fix
 Use "std::move(""drmHelper"")" instead of "drmHelper".

</details>

Vinish100 replied:
Fix this

dp0000 replied:
fixeed

rdkcmf-jenkins commented on middleware/drm/DrmSessionManager.cpp:
```diff
@@ -486,7 +489,7 @@ DrmSession* DrmSessionManager::createDrmSession(int &err, std::shared_ptr<DrmHel
 		return nullptr;
 	}
-	code =this->AcquireLicenseCb(drmHelper, selectedSlot, cdmError,  (GstMediaType)streamType, metaDataPtr, false);
+	code =this->AcquireLicenseCb(responseCode, drmHelper, selectedSlot, cdmError,  (GstMediaType)streamType, metaDataPtr, false);
```
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
90b1950cfe3ace0984d1e3d966942643
NOT_PRESENT
-->

Coverity issue no longer present as of: undefined
<details>
<summary>Show issue</summary>


# Coverity Issue - Variable copied when it could be moved
"drmHelper" is passed-by-value as parameter to "std::shared_ptr<DrmHelper>::shared_ptr(std::shared_ptr<DrmHelper> const &) /*explicit =default*/", when it could be moved instead.

_Low Impact, CWE-none_
_COPY_INSTEAD_OF_MOVE_

## How to fix
 Use "std::move(""drmHelper"")" instead of "drmHelper".

</details>

dp0000 replied:
fixed

balasaraswathy-n replied:
Same as above

rdkcmf-jenkins commented on middleware/drm/DrmSessionManager.cpp:
```diff
@@ -403,7 +406,7 @@
 		else
 		{
-			drmSession = DrmSessionManager::createDrmSession(err, drmHelper, player, streamType, metaDataPtr);
+			drmSession = DrmSessionManager::createDrmSession(responseCode, err, drmHelper, player, streamType, metaDataPtr);
```
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
a9810b24cc813902835aaec538694b6c
NOT_PRESENT
-->

Coverity issue no longer present as of: undefined
<details>
<summary>Show issue</summary>


# Coverity Issue - Variable copied when it could be moved
"drmHelper" is passed-by-value as parameter to "std::shared_ptr<DrmHelper>::shared_ptr(std::shared_ptr<DrmHelper> const &) /*explicit =default*/", when it could be moved instead.

_Low Impact, CWE-none_
_COPY_INSTEAD_OF_MOVE_

## How to fix
 Use "std::move(""drmHelper"")" instead of "drmHelper".

</details>

dp0000 replied:
fixed

balasaraswathy-n replied:
Doesn't seem to be resolved

rdkcmf-jenkins commented on drm/AampDRMLicManager.cpp:
```diff
@@ -1299,13 +1343,129 @@
+		AAMPTuneFailure failure = (AAMPTuneFailure)err;
+		if(AAMP_TUNE_FAILURE_UNKNOWN != failure)
+		{
+			long responseCode = (long)responseCode;
```
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
3617f785c31b7d523eca724f99486687
NOT_PRESENT
-->

Coverity issue no longer present as of: undefined
<details>
<summary>Show issue</summary>


# Coverity Issue - Parse warning
declaration hides parameter "responseCode" (declared at line 1464)

_Low Impact, CWE-398_
_PW.PARAMETER_HIDDEN_



</details>

dp0000 replied:
removed

rdkcmf-jenkins commented on drm/AampDRMLicManager.cpp:
```diff
@@ -1299,13 +1343,129 @@
+		AAMPTuneFailure failure = (AAMPTuneFailure)err;
+		if(AAMP_TUNE_FAILURE_UNKNOWN != failure)
+		{
+			long responseCode = (long)responseCode;
```
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
746dc33e48971e7c9c290d3d661e9377
NOT_PRESENT
-->

Coverity issue no longer present as of: undefined
<details>
<summary>Show issue</summary>


# Coverity Issue - Uninitialized scalar variable
Using uninitialized value "responseCode".

_High Impact, CWE-457_
_UNINIT_



</details>

rdkcmf-jenkins commented on drm/AampDRMLicManager.cpp:
```diff
@@ -355,11 +396,11 @@
 	if (code == KEY_PENDING)
 	{
-		code = handleLicenseResponse(drmHelper, sessionSlot, cdmError, httpResponseCode, httpExtendedStatusCode, licenseResponse, eventHandle,  isLicenseRenewal);
+		code = handleLicenseResponse(responseCode, drmHelper, sessionSlot, cdmError, httpResponseCode, httpExtendedStatusCode, licenseResponse, eventHandle,  isLicenseRenewal);
```
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
5f09a482979fcd9e9d49bb6adb3c9649
NOT_PRESENT
-->

Coverity issue no longer present as of: undefined
<details>
<summary>Show issue</summary>


# Coverity Issue - Variable copied when it could be moved
"drmHelper" is passed-by-value as parameter to "std::shared_ptr<DrmHelper>::shared_ptr(std::shared_ptr<DrmHelper> const &) /*explicit =default*/", when it could be moved instead.

_Low Impact, CWE-none_
_COPY_INSTEAD_OF_MOVE_

## How to fix
 Use "std::move(""drmHelper"")" instead of "drmHelper".

</details>

Vinish100 replied:
fix this

dp0000 replied:
fixed

rdkcmf-jenkins commented on middleware/InterfacePlayerRDK.cpp:
```diff
@@ -472,11 +531,12 @@ void InterfacePlayerRDK::SetPauseOnStartPlayback(bool enable)
 {
 	InterfacePlayerRDK *pInterfacePlayerRDK = (InterfacePlayerRDK *)user_data;
+	InterfacePlayerPriv* privatePlayer = pInterfacePlayerRDK->GetPrivatePlayer();
 	if (pInterfacePlayerRDK)
```
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
b65726fd04c9209310ef1e9427d61421
NOT_PRESENT
-->

Coverity issue no longer present as of: undefined
<details>
<summary>Show issue</summary>


# Coverity Issue - Dereference before null check
Null-checking "pInterfacePlayerRDK" suggests that it may be null, but it has already been dereferenced on all paths leading to the check.

_Medium Impact, CWE-476_
_REVERSE_INULL_



</details>

Vinish100 replied:
Please fix this

dp0000 replied:
fixed

rdkcmf-jenkins commented on middleware/InterfacePlayerRDK.cpp:
```diff
@@ -1252,7 +1332,7 @@
-			gstPrivateContext->decoderHandleNotified = false;
+			interfacePlayerPriv->gstPrivateContext->decoderHandleNotified = false;
 		}
 		stream->format = GST_FORMAT_INVALID;
```
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
cc3d03b3b37d92b0e947325f1455fe9d
NOT_PRESENT
-->

Coverity issue no longer present as of: undefined
<details>
<summary>Show issue</summary>


# Coverity Issue - Check of thread-shared field evades lock acquisition
Thread1 sets "format" to a new value. Now the two threads have an inconsistent view of "format" and updates to fields correlated with "format" may be lost.

_High Impact, CWE-543_
_LOCK_EVASION_

## How to fix
 Guard the modification of "format" and the read used to decide whether to modify "format" with the same set of locks.

</details>

rdkcmf-jenkins commented on middleware/InterfacePlayerRDK.cpp:
```diff
@@ -1773,6 +1861,7 @@
 {
 	InterfacePlayerRDK* pInterfacePlayerRDK = (InterfacePlayerRDK *)_this;
+	InterfacePlayerPriv* privatePlayer = pInterfacePlayerRDK->GetPrivatePlayer();
 	if (pInterfacePlayerRDK)
```
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
7568a269a0b816e3656db6fb94ff650d
NOT_PRESENT
-->

Coverity issue no longer present as of: undefined
<details>
<summary>Show issue</summary>


# Coverity Issue - Dereference before null check
Null-checking "pInterfacePlayerRDK" suggests that it may be null, but it has already been dereferenced on all paths leading to the check.

_Medium Impact, CWE-476_
_REVERSE_INULL_



</details>

dp0000 replied:
fixed

rdkcmf-jenkins commented on middleware/InterfacePlayerRDK.cpp:
```diff
@@ -521,12 +592,13 @@
 {
 	InterfacePlayerRDK *pInterfacePlayerRDK = (InterfacePlayerRDK *)user_data;
+	InterfacePlayerPriv* privatePlayer = pInterfacePlayerRDK->GetPrivatePlayer();
 	if (pInterfacePlayerRDK)
```
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
dcb2bdfd0d07b3768b0c3b490224470e
NOT_PRESENT
-->

Coverity issue no longer present as of: undefined
<details>
<summary>Show issue</summary>


# Coverity Issue - Dereference before null check
Null-checking "pInterfacePlayerRDK" suggests that it may be null, but it has already been dereferenced on all paths leading to the check.

_Medium Impact, CWE-476_
_REVERSE_INULL_



</details>

Vinish100 replied:
please fix this

dp0000 replied:
fixed

rdkcmf-jenkins commented on middleware/InterfacePlayerRDK.cpp:
```diff
@@ -56,29 +57,28 @@
+mSourceSetupCV(), mScheduler(), callbackMap(), setupStreamCallbackMap(), mDrmSystem(NULL), mEncrypt(NULL), mDRMSessionManager(NULL)
 {
-	gstPrivateContext = new GstPlayerPriv();
+	interfacePlayerPriv = new InterfacePlayerPriv();
```
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
69351cb2955e2120b33e31d236a4b44f
NOT_PRESENT
-->

Coverity issue no longer present as of: undefined
<details>
<summary>Show issue</summary>


# Coverity Issue - Resource leak in object
Allocating memory by calling "new InterfacePlayerPriv".

_High Impact, CWE-401_
_CTOR_DTOR_LEAK_



</details>

dp0000 replied:
deletion added

rdkcmf-jenkins commented on middleware/InterfacePlayerRDK.cpp:
```diff
@@ -1365,30 +1445,30 @@
-	gstPrivateContext->videoMuted = false;
-	gstPrivateContext->subtitleMuted = false;
-	gstPrivateContext->audioVolume = 1.0;
+	interfacePlayerPriv->gstPrivateContext->audioMuted = false;
```
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
8df2dd462935a54e628498fbb12c7112
PRESENT
-->

# Coverity Issue - Data race condition
Accessing "this->interfacePlayerPriv->gstPrivateContext->audioMuted" without holding lock "GstPlayerPriv.volumeMuteMutex". Elsewhere, "GstPlayerPriv.audioMuted" is written to with "GstPlayerPriv.volumeMuteMutex" held 2 out of 3 times.

_Medium Impact, CWE-366_
_MISSING_LOCK_

rdkcmf-jenkins commented on middleware/InterfacePlayerRDK.cpp:
```diff
@@ -715,10 +790,11 @@
 {
 	InterfacePlayerRDK *pInterfacePlayerRDK = (InterfacePlayerRDK *)user_data;
+	InterfacePlayerPriv* privatePlayer = pInterfacePlayerRDK->GetPrivatePlayer();
 	if (pInterfacePlayerRDK)
```
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
0596248cc68cf99fdfe6af3569d8e6d5
NOT_PRESENT
-->

Coverity issue no longer present as of: undefined
<details>
<summary>Show issue</summary>


# Coverity Issue - Dereference before null check
Null-checking "pInterfacePlayerRDK" suggests that it may be null, but it has already been dereferenced on all paths leading to the check.

_Medium Impact, CWE-476_
_REVERSE_INULL_



</details>

rdkcmf-jenkins commented on middleware/InterfacePlayerRDK.cpp:
```diff
@@ -4850,28 +4971,16 @@
-	retvalue = this->SetupStream(streamId, (void*)this, manifestUrl);
+	GstMediaType mediaType = static_cast<GstMediaType>(streamId);
+	this->TriggerEvent(InterfaceCB::startNewSubtitleStream, mediaType);
+	retvalue = this->SetupStream(mediaType, (void*)this, manifestUrl);
```
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
900484ff8a70f30ce33dd9c73e270670
NOT_PRESENT
-->

Coverity issue no longer present as of: undefined
<details>
<summary>Show issue</summary>


# Coverity Issue - Variable copied when it could be moved
"manifestUrl" is passed-by-value as parameter to "std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::basic_string(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const &)", when it could be moved instead.

_Low Impact, CWE-none_
_COPY_INSTEAD_OF_MOVE_

## How to fix
 Use "std::move(""manifestUrl"")" instead of "manifestUrl".

</details>

rdkcmf-jenkins commented on middleware/InterfacePlayerRDK.cpp:
```diff
@@ -666,14 +739,15 @@
 {
 	InterfacePlayerRDK *pInterfacePlayerRDK = (InterfacePlayerRDK *)user_data;
+	InterfacePlayerPriv* privatePlayer = pInterfacePlayerRDK->GetPrivatePlayer();
 	if (pInterfacePlayerRDK)
```
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
2e4fa3aa18720f403b83acb5626db22a
NOT_PRESENT
-->

Coverity issue no longer present as of: undefined
<details>
<summary>Show issue</summary>


# Coverity Issue - Dereference before null check
Null-checking "pInterfacePlayerRDK" suggests that it may be null, but it has already been dereferenced on all paths leading to the check.

_Medium Impact, CWE-476_
_REVERSE_INULL_



</details>

dp0000 replied:
fixed

rdkcmf-jenkins commented on middleware/InterfacePlayerRDK.cpp:
```diff
@@ -686,22 +760,23 @@
 {
 	InterfacePlayerRDK *pInterfacePlayerRDK = (InterfacePlayerRDK *)user_data;
+	InterfacePlayerPriv* privatePlayer = pInterfacePlayerRDK->GetPrivatePlayer();
 	if (pInterfacePlayerRDK)
```
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
1eafa348f9641a42f2f5b89eeca20737
NOT_PRESENT
-->

Coverity issue no longer present as of: undefined
<details>
<summary>Show issue</summary>


# Coverity Issue - Dereference before null check
Null-checking "pInterfacePlayerRDK" suggests that it may be null, but it has already been dereferenced on all paths leading to the check.

_Medium Impact, CWE-476_
_REVERSE_INULL_



</details>

dp0000 replied:
fixed

rdkcmf-jenkins commented on middleware/InterfacePlayerRDK.cpp:
```diff
@@ -4515,6 +4628,7 @@
 {
 	InterfacePlayerRDK *pInterfacePlayerRDK = (InterfacePlayerRDK *)user_data;
+	InterfacePlayerPriv* privatePlayer = pInterfacePlayerRDK->GetPrivatePlayer();
 	if (pInterfacePlayerRDK)
```
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
d5cf957472b0ed9788dac436016f06c1
NOT_PRESENT
-->

Coverity issue no longer present as of: undefined
<details>
<summary>Show issue</summary>


# Coverity Issue - Dereference before null check
Null-checking "pInterfacePlayerRDK" suggests that it may be null, but it has already been dereferenced on all paths leading to the check.

_Medium Impact, CWE-476_
_REVERSE_INULL_



</details>

rdkcmf-jenkins commented on middleware/InterfacePlayerRDK.cpp:
```diff
@@ -4372,18 +4483,19 @@
 	bool isPlayerReady = false;
 
-	if(pInterfacePlayerRDK && pInterfacePlayerRDK->gstPrivateContext)
+	if(pInterfacePlayerRDK && privatePlayer->gstPrivateContext)
```
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
d2b25a6a30b12f0b405946e0f4872c40
NOT_PRESENT
-->

Coverity issue no longer present as of: undefined
<details>
<summary>Show issue</summary>


# Coverity Issue - Dereference before null check
Null-checking "pInterfacePlayerRDK" suggests that it may be null, but it has already been dereferenced on all paths leading to the check.

_Medium Impact, CWE-476_
_REVERSE_INULL_



</details>

Deepikasri1234 committed:
suspecting null pointer being sent in createdrm ssession

rdkcmf-jenkins commented on drm/AampDRMLicManager.cpp:
```diff
@@ -185,9 +225,18 @@ void AampDRMLicenseManager::renewLicense(std::shared_ptr<DrmHelper> drmHelper, v
+                //std::shared_ptr<void> e = cdmidecryptor->sessionManager->DrmMetaDataCb();
+        }
+
 	DrmMetaDataEventPtr* eventHandlePtr = static_cast<DrmMetaDataEventPtr*>(metaDataPtr);
```
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
9ad413ce73996c71028b143ed5d59183
NOT_PRESENT
-->

Coverity issue no longer present as of: undefined
<details>
<summary>Show issue</summary>


# Coverity Issue - Pointer to local outside scope
Using "metaDataPtr", which points to an out-of-scope variable "drmEvent".

_High Impact, CWE-562_
_RETURN_LOCAL_



</details>

Deepikasri1234 committed:
removed metadata

Deepikasri1234 committed:
debug prints added

Deepikasri1234 committed:
debug prints added

Deepikasri1234 committed:
analaysing crash

Deepikasri1234 committed:
meta data ptr updated in plugin code

Deepikasri1234 committed:
RDK-56050 Merging of OCDM, gst-plugins-rdk into Library - Design

Reason for change: review comments addressed
Test Procedure: Refer Ticket

Risks: Low

Priority: P1

Signed-off-by: Deepikasri N <Deepikasri_N@Comcast.Com

Deepikasri1234 committed:
RDK-56050 Merging of OCDM, gst-plugins-rdk into Library - Design

Reason for change: review comments adressed
Test Procedure: Refer Ticket

Risks: Low

Priority: P1

Signed-off-by: Deepikasri N <Deepikasri_N@Comcast.Com

Deepikasri1234 committed:
RDK-56050 Merging of OCDM, gst-plugins-rdk into Library - Design

Reason for change: review comments addresed
Test Procedure: Refer Ticket

Risks: Low

Priority: P1

Signed-off-by: Deepikasri N <Deepikasri_N@Comcast.Com

dp0000 committed:
Merge branch 'dev_sprint_25_2' into feature/RDK-56050

rdkcmf-jenkins commented:
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
b847c24d2d0738430234b752556c59dd
PRESENT
-->

# Coverity Issue - Variable copied when it could be moved
"seiTimecode" is passed-by-value as parameter to "std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::basic_string(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const &)", when it could be moved instead.

_Low Impact, CWE-none_
_COPY_INSTEAD_OF_MOVE_

## How to fix
 Use "std::move(""seiTimecode"")" instead of "seiTimecode".

## Issue location
This issue was discovered outside the diff for this Pull Request. You can find it at:
[AampEvent.cpp:163](undefined//blob/undefined/AampEvent.cpp#L163)

Deepikasri1234 committed:
RDK-56050 Merging of OCDM, gst-plugins-rdk into Library - Design

Reason for change: for dev_Sprint_25_2
Test Procedure: Refer Ticket

Risks: Low

Priority: P1

Change-Id: If0a31d066478806bae29f09ab829b1108d7856bc
Signed-off-by: Deepikasri N <Deepikasri_N@Comcast.Com

Deepikasri1234 committed:
Merge branch 'dev_sprint_25_2' into feature/RDK-56050

Deepikasri1234 committed:
Merge branch 'dev_sprint_25_2' into feature/RDK-56050

Vinish100 commented on CMakeLists.txt:
```diff
@@ -18,7 +18,17 @@
 # XCode schema generation / flag setting
 # Other flags see here: https://cmake.org/cmake/help/latest/prop_tgt/XCODE_GENERATE_SCHEME.html
 cmake_minimum_required (VERSION 3.5)
+set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -g")
```
Why is this required?

dp0000 replied:
removed

Vinish100 replied:
Still seeing this code here

balasaraswathy-n replied:
Resolve this

Vinish100 commented on CMakeLists.txt:
```diff
@@ -479,28 +362,7 @@ if (CMAKE_TELEMETRY_2_0_REQUIRED)
-	middleware/subtec/libsubtec/SubtecChannel.cpp
-)
-
-add_library(subtec SHARED ${AAMP_SUBTEC_SOURCES})
```
add_library is missing. So what is the use of all AAMP_SUBTEC_PUBLIC_HEADERS variables

Vinish100 replied:
and all other SUBTEC related variables

dp0000 replied:
removed

Vinish100 commented on AampConfig.cpp:
```diff
@@ -863,6 +863,10 @@ void AampConfig::ApplyDeviceCapabilities()
 	bool isSecMgr = isSecManagerEnabled();
 	SetConfigValue(AAMP_DEFAULT_SETTING, eAAMPConfig_UseSecManager, isSecMgr);
+	
+	bool isGstSubtec = SocUtils::isGstSubtecEnabled();
```
Shouldn't this be enabled for all RDK platforms and disabled only for simulators?

varatharajan568 replied:
It is not enabled for all RDK platforms by default. It will only be enabled if the following flag is set in the .bb file:
CMAKE_GST_SUBTEC_ENABLED.

Vinish100 commented on AampDRMLicPreFetcher.cpp:
```diff
@@ -460,8 +460,8 @@ void AampLicensePreFetcher::NotifyDrmFailure(LicensePreFetchObjectPtr fetchObj,
 	bool ret = false;
-
-	DrmMetaDataEventPtr e = std::make_shared<DrmMetaDataEvent>(AAMP_TUNE_FAILURE_UNKNOWN, "", 0, 0, mIsSecClientError, mPrivAAMP->GetSessionId());
+	bool isSecClientError = isSecFeatureEnabled();
```
Have a follow up ticket to remove these, does AAMP need to be aware of if the request is made through SecClient, SecManager or CPS?

dp0000 replied:
can be taken as follow up ticket

balasaraswathy-n replied:
Kindly link the follow up ticket

dp0000 replied:
https://ccp.sys.comcast.net/browse/RDK-57601. Captured here

Vinish100 commented on drm/AampDRMLicManager.cpp:
```diff
@@ -50,31 +50,61 @@
+
+     /**  Register the SetFailure callback for TriggerSetFailureCb */
+      instance->RegisterSetFailure([_this](void* ptr, int err){
+		      _this->TriggerSetFailure(ptr,err);
```
Fix indentation here and everywhere else

dp0000 replied:
fixed alignment

dp0000 replied:
fixed

Vinish100 commented on drm/AampDRMLicManager.cpp:
```diff
@@ -166,8 +197,8 @@ void AampDRMLicenseManager::renewLicense(std::shared_ptr<DrmHelper> drmHelper, v
 			mLicenseRenewalThreads[sessionSlot] = std::thread([this, drmHelper, sessionSlot, aampInstance] {
-        this->licenseRenewalThread(drmHelper, sessionSlot, aampInstance);
-    });
+					this->licenseRenewalThread(drmHelper, sessionSlot, aampInstance);
```
Fix indentation

dp0000 replied:
fixed

Vinish100 commented on drm/AampDRMLicManager.cpp:
```diff
@@ -1280,12 +1315,19 @@ DrmData * AampDRMLicenseManager::getLicenseSec(const LicenseRequest &licenseRequ
 	}
-	UpdateLicenseMetrics(DRM_GET_LICENSE_SEC, *httpCode, licenseRequest.url.c_str(), downloadTimeMS, eventHandle, nullptr );
+				
+        UpdateLicenseMetrics(DRM_GET_LICENSE_SEC, *httpCode, licenseRequest.url.c_str(), downloadTimeMS, eventHandle, nullptr );				
```
Fix indentation

dp0000 replied:
fixed

Vinish100 commented on drm/AampDRMLicManager.cpp:
```diff
@@ -1299,13 +1341,101 @@ void AampDRMLicenseManager::ProfilerUpdate()
+	e->setFailure((AAMPTuneFailure)err);
+}
+
+std::shared_ptr<void> AampDRMLicenseManager::TriggerDrmMetaDataEvent()
```
Why is this required?

dp0000 replied:
required for sending metadataevent to plugin

Vinish100 replied:
@dp0000 This needs to be reworked. The DRM metadata can be considered as a structure to pull additional details from licensemanager. The event should be triggered in AAMP. The entire flow needs to be reworked
from dp0000: RDK-59416 - captured here and can be work as the seperate ticket.

Vinish100 commented on install-aamp.sh:
```diff
@@ -161,7 +150,6 @@ fi
 fi
-
 if [ ${OPTION_SUBTEC_SKIP} = false ] ; then 
     subtec_install_build_fn "${CLEAN}"
```
Installing subtec should also be a middleware dependency. Please have a follow up ticket to move to middleware

varatharajan568 replied:
Follow up ticket to move to middleware : [RDK-58972](https://ccp.sys.comcast.net/browse/RDK-58972)

pstroffolino committed:
Merge branch 'dev_sprint_25_2' into feature/RDK-56050

Deepikasri1234 committed:
Merge branch 'dev_sprint_25_2' into feature/RDK-56050

Vinish100 commented on AampEvent.cpp:
```diff
@@ -1036,7 +1036,7 @@ void DrmMetaDataEvent::ConvertToVerboseErrorCode(int32_t httpCode, int32_t httpE
 	mSecManagerReasonCode = CONTENT_SECURITY_MANAGER_REASON_DRM_GENERAL_FAILURE;
 	//look for the correct code from the lookup
-	if (getAsVerboseErrorCode(httpCode, mSecManagerClass, mSecManagerReasonCode)) 
+	if (getAsVerboseErrorCode(httpCode, mSecManagerClass, mSecManagerReasonCode))
```
This should also eventually move to middleware

Vinish100 commented on aampgstplayer.cpp:
```diff
@@ -411,7 +411,8 @@ AAMPGstPlayer::AAMPGstPlayer(PrivateInstanceAAMP *aamp, id3_callback_t id3Handle
 		InitializePlayerConfigs(this,playerInstance);
 		playerInstance->SetPlayerName(PLAYER_NAME);
-		playerInstance->setEncryption((void*)aamp);
+                playerInstance->setEncryption((void*)aamp, (void*)aamp->mDRMLicenseManager->mDrmSessionManager);
```
Fix indentation

dp0000 replied:
fixed

Vinish100 replied:
Still not aligned

Vinish100 commented on middleware/CMakeLists.txt:
```diff
@@ -16,9 +17,363 @@
+set(SUBTEC_COMPILE_FLAGS "${CMAKE_CXX_FLAGS} -DSUBTEC_PACKET_DEBUG=1")
+if(CMAKE_PLATFORM_UBUNTU)
+	message("CMAKE_PLATFORM_UBUNTU set")
+	set(LIBPLAYERGSTINTERFACE_DEFINES "${LIBPLAYERGSTINTERFACE_DEFINES} -DUBUNTU=1 -DNO_NATIVE_AV=1")
```
NO_NATIVE_AV macro is not present anymore. Please rebase and check other macros as well

dp0000 replied:
fixed

Vinish100 commented on middleware/InterfacePlayerPriv.h:
```diff
@@ -0,0 +1,327 @@
+#define GST_ERROR_DESCRIPTION_LENGTH 256
+#define NOW_STEADY_TS_MS std::chrono::duration_cast<std::chrono::milliseconds>(std::chrono::steady_clock::now().time_since_epoch()).count() /**< Getting current steady clock in milliseconds */
+
+typedef enum
```
Have a follow-up ticket to remove these from Middleware

dp0000 replied:
noted

Vinish100 replied:
@dp0000  Please post the follow up ticket

by Dp0000: Captured in  RDKEMW-8602

Vinish100 commented on middleware/InterfacePlayerPriv.h:
```diff
@@ -0,0 +1,327 @@
+	GstEOS_INJECTION_MODE_STOP_ONLY,
+};
+
+typedef enum
```
Same for State enums

Vinish100 replied:
@dp0000  Please remove them from middleware and have a follow-up ticket created

by dp0000: Already captured in RDKEMW-8602

Vinish100 commented on middleware/InterfacePlayerRDK.cpp:
```diff
@@ -92,12 +92,21 @@ InterfacePlayerRDK::~InterfacePlayerRDK()
-		pthread_mutex_destroy(&gstPrivateContext->stream[i].sourceLock);
+		pthread_mutex_destroy(&interfacePlayerPriv->gstPrivateContext->stream[i].sourceLock);
 	}
+	pthread_mutex_destroy(&mProtectionLock);
```
Add delete interfacePlayerPriv;

dp0000 replied:
fixed

Vinish100 commented on middleware/InterfacePlayerRDK.cpp:
```diff
@@ -345,17 +381,40 @@ void InterfacePlayerRDK::ConfigurePipeline(int format, int audioFormat, int auxF
 		stream->eosReached = false;
 		stream->firstBufferProcessed = false;
 	}
+#if 0 
```
Please remove this #if0 code

Vinish100 commented on middleware/InterfacePlayerRDK.cpp:
```diff
@@ -634,8 +707,8 @@ void MonitorAV( InterfacePlayerRDK *pInterfacePlayerRDK )
 							   monitorAVState->description,
-							   monitorAVState->av_position[eGST_MEDIATYPE_VIDEO],
-							   monitorAVState->av_position[eGST_MEDIATYPE_AUDIO],
+							   (gint64)monitorAVState->av_position[eGST_MEDIATYPE_VIDEO],
```
Why typecasting? Its already gint64

dp0000 replied:
suppresed warnings

Vinish100 commented on middleware/InterfacePlayerRDK.cpp:
```diff
@@ -2325,27 +2421,27 @@ void InterfacePlayerRDK::SendGstEvents(GstMediaType mediaType, GstClockTime pts)
  * @ret TRUE if override is enabled, FALSE otherwise
  */
-gboolean InterfacePlayerRDK::SendQtDemuxOverrideEvent(GstMediaType mediaType, GstClockTime pts, const void *ptr, size_t len)
+gboolean InterfacePlayerPriv::SendQtDemuxOverrideEvent(int mediaType, GstClockTime pts, bool enablePTSReStamp, int vodTrickModeFPS, const void *ptr, size_t len)
```
Preferably move InterfacePlayerPriv impl to a different file

dp0000 replied:
can be taken in seperate ticket

Vinish100 replied:
@dp0000 Please share new ticket number


by dp0000: All the gstreamer related deprecations are captured under https://ccp.sys.comcast.net/browse/RDKEMW-8602

Vinish100 commented on middleware/InterfacePlayerRDK.h:
```diff
@@ -27,33 +27,47 @@
+	long long tLastReported;
+	long long tLastSampled;
+	const char *description;
+	signed long av_position[2];
```
use int64_t for these data types and long long

dp0000 replied:
UPDATED

Vinish100 commented on middleware/PlayerUtils.h:
```diff
@@ -92,6 +92,23 @@ void ResolveURL(std::string& dst, std::string base, const char *uri , bool bProp
+/**
+ * @brief Resolve file URL from the base and file path
+ */
+void player_ResolveURL(std::string& dst, std::string base, const char *uri , bool bPropagateUriParams);
```
Not seeing it being used anywhere. Please check if required

dp0000 replied:
removed

balasaraswathy-n replied:
Still seen in the code kindly remove

Vinish100 commented on middleware/PlayerUtils.h:
```diff
@@ -92,6 +92,23 @@ void ResolveURL(std::string& dst, std::string base, const char *uri , bool bProp
+ * @param[in] uri manifest/ fragment uri
+ * @retval return pointer just past protocol (i.e. http://) if present (or) return NULL uri doesn't start with protcol
+ */
+static const char * ParseUriProtocol(const char *uri);
```
Not seeing it being used anywhere. Please check if required

dp0000 replied:
removed

balasaraswathy-n replied:
Not yet removed

Vinish100 commented on middleware/drm/DrmSessionManager.h:
```diff
@@ -403,16 +403,57 @@ class DrmSessionManager
+	void RegisterMetaDataCb(const DrmMetaDataCallback callback)
+	{
+		DrmMetaDataCb = callback;
+	};
```
Trailing semi-colon not required?

balasaraswathy-n replied:
Not yet addressed

Vinish100 commented on middleware/gst-plugins/gst_subtec/gstvipertransform.cpp:
```diff
@@ -589,10 +585,10 @@ plugin_init (GstPlugin * plugin)
 #endif
 #ifndef PACKAGE_NAME
-#define PACKAGE_NAME "AAMPGstreamerPlugins"
+#define PACKAGE_NAME "GstreamerPlugins"
```
RDKGStreamerPlugins or MWGStreamerPlugins would be nice

dp0000 replied:
renamed

balasaraswathy-n replied:
Not renamed, kindly close

Deepikasri1234 committed:
alignment missed in licmanager

rdkcmf-jenkins commented on drm/AampDRMLicManager.cpp:
```diff
@@ -49,32 +49,62 @@
-     });
+	/* Register the callback for acquire license data */
+	instance->RegisterLicenseDataCb([_this](int &responseCode, std::shared_ptr<DrmHelper> drmHelper, int sessionSlot, int &cdmError, int streamType,void *metaDataPtr, bool isLicenseRenewal = false ) -> KeyState {
+			return _this->acquireLicense(responseCode, drmHelper, sessionSlot, cdmError,
```
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
398e7589e469827525758a9c4292cac3
PRESENT
-->

# Coverity Issue - Variable copied when it could be moved
"drmHelper" is passed-by-value as parameter to "std::shared_ptr<DrmHelper>::shared_ptr(std::shared_ptr<DrmHelper> const &) /*explicit =default*/", when it could be moved instead.

_Low Impact, CWE-none_
_COPY_INSTEAD_OF_MOVE_

## How to fix
 Use "std::move(""drmHelper"")" instead of "drmHelper".

Deepikasri1234 committed:
addressed Part-2 review comments

Commit:
remove NO_NATIVE_AV

Deepikasri1234 committed:
rebasing changes

Deepikasri1234 committed:
subtec review comments addressed

Deepikasri1234 committed:
Merge branch 'dev_sprint_25_2' into feature/RDK-56050

varatharajan568 committed:
Merge branch 'dev_sprint_25_2' into feature/RDK-56050

varatharajan568 committed:
Update InterfacePlayerRDK.cpp

Modified the GST Plugin name by adding the aamp prefix

varatharajan568 committed:
Added debug to ensure DRM BUILD PROFILE

Added debug to ensure DRM BUILD PROFILE

varatharajan568 committed:
Update gstwidevinedecryptor.h

Updated the widevine decryptor plugin name

varatharajan568 committed:
Update gstplayreadydecryptor.h

Updated the playready decryptor plugin name

varatharajan568 committed:
Update gstclearkeydecryptor.h

Updated the clearkey decryptor plugin name

varatharajan568 committed:
Update gstverimatrixdecryptor.h

Updated the verimatrix decryptor plugin name

Vinish100 commented on CMakeLists.txt:
```diff
@@ -322,7 +321,7 @@ set(LIBAAMP_SOURCES
 	dash/utils/StringEx.cpp
 	dash/utils/Path.cpp
-	ProcessHandler.cpp ProcessHandler.h
+	#ProcessHandler.cpp ProcessHandler.h
```
Remove if not required

Vinish100 replied:
What about these?

Deepikasri1234 committed:
Merge branch 'dev_sprint_25_2' into feature/RDK-56050

rekhap2kandhavelan committed:
Update CMakeLists.txt

resolved error

Deepikasri1234 committed:
renamed gstplugin to gstaamp

Deepikasri1234 committed:
Merge branch 'dev_sprint_25_2' into feature/RDK-56050

Deepikasri1234 committed:
renamed plugin name back to "aamp" from "plugin"
Created playback issue in vipa and so reverted

balasaraswathy-n committed:
Update gstwidevinedecryptor.cpp

Deepikasri1234 committed:
resolved rebasing conflicts

Deepikasri1234 committed:
resolving conflicts

Deepikasri1234 committed:
resolving rebsning conflicts

Deepikasri1234 committed:
while rebasing;PluginTest got automerged in cmake and causing cmake error .. removed as it is mved to middlware

Deepikasri1234 committed:
removed unused endif

Deepikasri1234 committed:
Merge branch 'dev_sprint_25_2' into feature/RDK-56050

Deepikasri1234 committed:
Merge branch 'dev_sprint_25_2' into feature/RDK-56050

dp0000 committed:
Merge branch 'dev_sprint_25_2' into feature/RDK-56050

Vinish100 commented on drm/AampDRMLicManager.h:
```diff
@@ -252,16 +258,32 @@ class AampDRMLicenseManager
+	 * @param[in] StreamType
+	 * @return void */
+        void TriggerLAProfileBeginCb(int streamType);
+        void TriggerLAProfileEndCb(int streamType);
```
Missing function headers for these new functions

dp0000 replied:
ADDED

Deepikasri1234 committed:
addressing review comments

Deepikasri1234 committed:
renaming in other files

Deepikasri1234 committed:
long long to int64_t

Deepikasri1234 committed:
if 0 removed

rdkcmf-jenkins commented:
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
cc3d03b3b37d92b0e947325f1455fe9d
PRESENT
-->

# Coverity Issue - Check of thread-shared field evades lock acquisition
Thread1 sets "format" to a new value. Now the two threads have an inconsistent view of "format" and updates to fields correlated with "format" may be lost.

_High Impact, CWE-543_
_LOCK_EVASION_

## How to fix
 Guard the modification of "format" and the read used to decide whether to modify "format" with the same set of locks.

## Issue location
This issue was discovered outside the diff for this Pull Request. You can find it at:
[middleware/InterfacePlayerRDK.cpp:1346](undefined//blob/undefined/middleware/InterfacePlayerRDK.cpp#L1346)

rdkcmf-jenkins commented:
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
900484ff8a70f30ce33dd9c73e270670
PRESENT
-->

# Coverity Issue - Variable copied when it could be moved
"manifestUrl" is passed-by-value as parameter to "std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::basic_string(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const &)", when it could be moved instead.

_Low Impact, CWE-none_
_COPY_INSTEAD_OF_MOVE_

## How to fix
 Use "std::move(""manifestUrl"")" instead of "manifestUrl".

## Issue location
This issue was discovered outside the diff for this Pull Request. You can find it at:
[middleware/InterfacePlayerRDK.cpp:5028](undefined//blob/undefined/middleware/InterfacePlayerRDK.cpp#L5028)

rdkcmf-jenkins commented on middleware/InterfacePlayerRDK.cpp:
```diff
@@ -1397,7 +1462,7 @@ void InterfacePlayerRDK::SetPendingSeek(bool state)
 	{
-		gstPrivateContext->stream[i].pendingSeek = state;
+		interfacePlayerPriv->gstPrivateContext->stream[i].pendingSeek = state;
 	}
```
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
8df2dd462935a54e628498fbb12c7112
PRESENT
-->

# Coverity Issue - Data race condition
Accessing "this->interfacePlayerPriv->gstPrivateContext->audioMuted" without holding lock "GstPlayerPriv.volumeMuteMutex". Elsewhere, "GstPlayerPriv.audioMuted" is written to with "GstPlayerPriv.volumeMuteMutex" held 2 out of 3 times.

_Medium Impact, CWE-366_
_MISSING_LOCK_

dp0000 replied:
no where this variable is being held and so its not updated under mutexes

Deepikasri1234 committed:
Godspell check fixed

Deepikasri1234 committed:
rebasing changes conflicts

Deepikasri1234 committed:
addressing god spell errors

Deepikasri1234 committed:
alignment handling

aahmed878 committed:
Resolve compilation error in MAC

rkandh015 committed:
resolved compilation error inubuntu for L1

Vinish100 commented on middleware/gst-plugins/COPYING.LGPL:
```diff
@@ -9,7 +9,7 @@ Everyone is permitted to copy and distribute verbatim copies
 
 [This is the first released version of the Lesser GPL.  It also counts
- as the successor of the GNU Library Public License, version 2, hence
+ as the successor of the Public License, version 2, hence
```
@dp0000 Why is there a change here?

dp0000 replied:
reverted back

Deepikasri1234 committed:
Merge branch 'dev_sprint_25_2' into feature/RDK-56050

Deepikasri1234 committed:
coverity issues fixed

Deepikasri1234 committed:
license files got changed , reverting

pstroffolino commented on .github/workflows/L1-tests.yml:
```diff
@@ -50,4 +50,4 @@ jobs:
         if: failure()
-        run: tail -n 50 test/utests/build/ctest-results.xml
\ No newline at end of file
+        run: tail -n 50 test/utests/build/ctest-results.xml
```
why is this line with no obvious change listed in review?

Deepikasri1234 committed:
Merge branch 'dev_sprint_25_2' into feature/RDK-56050

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/rdkcentral/aamp?pullRequest=369) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/rdkcentral/aamp?pullRequest=369) before we can accept your contribution.<br/>**1** out of **3** committers have signed the CLA.<br/><br/>:white_check_mark: aahmed878<br/>:x: Deepikasri1234<br/>:x: rkandh015<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/rdkcentral/aamp?pullRequest=369) it.</sub>
